themes/default/content/docs/guides/organizing-projects-stacks/_index.md

```diff
@@ -177,3 +275,101 @@ These component resources can be packaged up and stored alongside all of your ot
+
+```typescript
+// index.ts
+import * as util from "./util";
+...
+const forwarderHandle = util.forwardPrometheusService(p8sService, p8sDeployment, {
+    localPort,
+});
+```
+
+```typescript
+// utils.ts
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+
+export function forwardPrometheusService(
+    service: pulumi.Input<k8s.core.v1.Service>,
+    deployment: pulumi.Input<k8s.extensions.v1beta1.Deployment>,
+    opts: PromPortForwardOpts,
+): pulumi.Output<() => void> {
+    if (pulumi.runtime.isDryRun()) {
+        return pulumi.output(() => undefined);
+    }
+
+    return pulumi.all([service, deployment]).apply(([s, d]) => pulumi.all([s.metadata, d.urn])).apply(([meta]) => {
+        return new Promise<() => void>((resolve, reject) => {
+            const forwarderHandle = spawn("kubectl", [
+                "port-forward",
+                `service/${meta.name}`,
+                `${opts.localPort}:${opts.targetPort || 80}`,
+            ]);
+
+            forwarderHandle.stdout.on("data", data => resolve(() => forwarderHandle.kill()));
+            forwarderHandle.stderr.on("data", data => reject());
+        });
+    });
+}
+
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+Organize your code in a way that makes it easy to understand and maintain. One way to do this in Go is to break out your code into separate files, and then import them into your main file. In this example, the entrypoint for our Pulumi program is `main.go`, but we use the `utils.go` file for supporting functions.
+
+```go
+// main.go
+package main
+import (
+  "github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+  pulumi.Run(func(ctx *pulumi.Context) error {
+
+        _, err := forwardPrometheusService(ctx)
+    if err != nil {
+      return err
+    }
+
+  }
+}
+```
+
+```go
+// utils.go
+package main
+import (
+  appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/apps/v1"
+  corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
+  metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
+  "github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+  "github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+)
+
+func forwardPrometheusService(ctx *pulumi.Context)  {
+...
+
+  return nil
+}
+
+```
+
+{{% /choosable %}}
+
+There are a couple of reasons that this pattern is helpful. One, in this particular case, is that the `forwardPrometheusService` function exists to forward the Prometheus service to localhost, so we can check it. If you are running in-cluster, we probably don't need it! So we could add a conditional to determine if we need to run that function - which makes our code a lot clearer.
```

Comment:
To be fair, though, you could put that same conditional into `index.ts` or `main.go`. Since the point seems to be to clarify the benefit of breaking utility code (for example) out into separate modules or files within a package, it might be clearer to have the example be a utility function that does something super simple, like format a string in a way that's commonly used across a team or organization or something. What do you think? Otherwise, you sort of have to imagine (particularly if the body of the function is empty) what the `forwardPrometheusService` is doing and why it'd be a good candidate for extraction into a `utils` module.